### PR TITLE
feat: `Switch` atom

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- add new `Switch`component using Radix primitives

### CODE REVIEW

File: src/components/atoms/switch/index.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from "react"
+import clsx from "clsx"
+import * as _Switch from "@radix-ui/react-switch"
```
Comment (pKorsholm) [inline src/components/atoms/switch/index.tsx:3]: Nit: Rename this to `RadixSwitch` as we have done in other components that Radix components

File: src/components/atoms/switch/index.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * A controlled switch component atom.
+ */
+function Switch(props: _Switch.SwitchProps) {
+  const { checked } = props
+
+  return (
+    <_Switch.Root
+      {...props}
+      className={clsx("w-8 h-[18px] rounded-full", {
```
Comment (pKorsholm) [inline src/components/atoms/switch/index.tsx:14]: We could also add a transition to the background color as well

We could consider adding this to the `tailwind.config.ts` as well in the `transitionProperty`s. I'm thinking we could add a `transition-bg` class for background color animation: 
```javascript
 transitionProperty: {
        ...
        bg: "background-color",
      },
```
 wdyt?

File: src/components/atoms/switch/index.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        "bg-gray-300": !checked,
+        "bg-violet-60": checked,
+      })}
+    >
+      <_Switch.Thumb
+        className={clsx("w-2 h-2 bg-white rounded-full block transition-all", {
```
Comment (pKorsholm) [inline src/components/atoms/switch/index.tsx:20]: nit: we could limit the transition class to just translations here

**Review by pKorsholm — COMMENTED** [review]
Love the animation of the "thumb", very nice.

I just have a couple of comments 😄

**Review by srindom — APPROVED** [review]
LGTM!

**Pull request merged**